app/src/main/java/com/example/mystore/ui/home/filters/ExpandableList.kt

```diff
@@ -47,13 +47,11 @@ fun ExpandableList(
     onFiltersChanged: (CategoryDetailsModel) -> Unit
 ) {
     var expanded by remember { mutableStateOf(false) }
-    val selectedCategoriesString: String = stringResource(id = R.string.all_categories_filter)
-    val filteredCategories2: List<CategoryDetailsModel> =
-        data.filter { selectedCategoriesString.contains(it.name) }
-    if (filteredCategories2.isNotEmpty() && filteredCategories2.count() != data.count()) {
-        filteredCategories2.joinToString { it.name }
-    } 
-
+    var selectedCategoriesString = stringResource(id = R.string.all_categories_filter)
+    val filteredCategories = data.filter { selectedCategories.contains(it.name) }
```

Comment:
Make it into a getter.

Reply:
`filteredCategories` here is a property (as there are no Java-like fields in Kotlin, only properties). So technically it has both a setter and a getter internally.

A custom getter is evaluated upon usage only, however. Here you will be filtering this data and assigning a value on every single rerender.